src/classes/CON_Email.cls

```diff
@@ -0,0 +1,145 @@
+/*
+    Copyright (c) 2017 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+/**
+* @author Salesforce.org
+* @date 2017
+* @group Contacts
+* @group-content ../../ApexDocContent/Contacts.htm
+* @description Handles processing of COntact preferred emails. Copies custom email fields to the standard field.
+*
+*/
+public class CON_Email {
+
+    /*******************************************************************************************************
+    * @description Sets the Email field depending on the value of Preferred Email
+    * @param Contact the contact to update
+    * @param oldlist list of old contacts from update or delete context of a trigger
+    */
+	public static void processPreferredEmail(Contact contact, List<Contact> oldlist) {
+
+        // No need to run if we are in a delete context
+        if(contact == null) {
+            return;
+        }
+
+		Boolean isUpdateOrDelete = (oldlist != null) ? true : false;
+
+	    // Store default email field mappings to preferred email pick list.
+	    Map<String,String> preferredEmailMappings = new Map<String,String>{   
```

Comment:
As I see it, we cannot take out the mappings because the values available in the Preferred Email field are not directly related to the actual custom fields. We could possibly use `string.startsWith(prefix)` but it is not a perfect 1:1 mapping of field labels and preferred email options.